chains/evm/deposithandlers/erc20.go

```diff
@@ -0,0 +1,62 @@
+package deposithandlers
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ChainSafe/sygma-core/types"
+)
+
+type Erc20DepositHandler struct {
+	ArbitraryFunction arbitraryFunction
```

Comment:
Remove the function and the config for all functions that don't need it.
We just need the struct and then the arbitrary functionality is more in the sense that it can be implemented per handler with specific properties.